### PR TITLE
Add optional length constraints.

### PR DESCRIPTION
**Pull request by toddthomas**
`RandomWord` is often used to obtain names for factory-built objects. Name attributes commonly have constraints on their length enforced by `ActiveRecord` validations or similar mechanisms.

Some of `RandomWord`'s "words" are actually snake-case phrases, and can be surprisingly long. The longest noun is "american_federation_of_labor_and_congress_of_industrial_organizations", with 69 characters. The longest adjective is "naked_as_the_day_you_were_born", with 30 characters.

Thus the odds that `RandomWord` will provide a word that causes a maximum length validation failure are greater than one might have assumed. I ran into this problem with an attribute constrained to 75 characters. That seems quite long, but the factory prefixed the random word with 19 additional characters, and `RandomWord` has 7 words longer than 56 characters.

I expanded the `.nouns` and `.adjs` APIs to allow constraining the length of the words returned. Examples:
```ruby
RandomWord.nouns(not_longer_than: 56).next
RandomWord.adjs(not_shorter_than: 3).next
RandomWord.adjs(not_shorter_than: 16, not_longer_than: 672).next
```

You may quibble with the negative sense of the constraints. My reasoning: attribute length validations are expressed in terms of the minimum and/or maximum permitted length. For example:
```ruby
validates_length_of :first_name, :maximum => 30
```
I prefer saying
```ruby
RandomWord.nouns(not_longer_than: 30).next
```
to
```ruby
RandomWord.nouns(shorter_than_or_equal_to: 30).next
```
or
```ruby
RandomWord.nouns(shorter_than: 31).next
```

### CODE REVIEW

File: lib/random_word.rb [hunk 1]
```diff
@@ -9,22 +9,35 @@
...
+        skipped.clear
       end
 
     rescue OutOfWords
       # we are done.
     end
 
+    def set_constraints(opts)
+      @min_length = opts[:not_shorter_than] || 1
+      @max_length = opts[:not_longer_than] || nil
```
Comment (pezra) [inline lib/random_word.rb]: s/`nil`/`Float::INFINITY`

File: lib/random_word.rb [hunk 2]
```diff
@@ -69,8 +88,10 @@ def each()
     # Create a random, non-repeating enumerator for a list of words
     # (or anything really).
     def enumerator(word_list, list_of_regexs_or_strings_to_exclude = [])
+      @word_list = word_list
       word_list.extend EachRandomly
       word_list.random_word_exclude_list = list_of_regexs_or_strings_to_exclude
+      word_list.set_constraints(not_shorter_than: 1, not_longer_than: nil)
```
Comment (pezra) [inline lib/random_word.rb]: Maybe defaulting the constraints should happen in the `.extended` hook?

**Comment by pezra** [comment]
fwiw, i definitely wouldn't write this as a module with which to extend arrays today. past me has deeply questionable judgment. 😃

File: lib/random_word.rb [hunk 3]
```diff
@@ -9,22 +9,35 @@
...
     def each_randomly(&blk)
       used = Set.new
+      skipped = Set.new
       exclude_list = Array(@random_word_exclude_list)
       while true
         idx = next_unused_idx(used)
         used << idx
         word = at(idx)
-        next if exclude_list.any?{|r| r === word }
+        if exclude_list.any? { |r| r === word } || word.length < min_length || (!(max_length.nil?) && word.length > max_length)
```
Comment (toddthomas) [inline lib/random_word.rb]: I agree. Will do.

File: lib/random_word.rb (hunk 1 above)
Comment (toddthomas) [inline lib/random_word.rb]: I prefer doing something like this, but didn't know this in particular was an option. I was thinking too much about integers, in which case I don't like using `MAX_INT` or whatever that is in Ruby. Do you consider this to be idiomatic Ruby? It makes me feel a little funny, but I could get used to it.

File: lib/random_word.rb (hunk 2 above)
Comment (toddthomas) [inline lib/random_word.rb]: I like it.

**Comment by toddthomas** [comment]
No worries; it's code that is so old that it's probably protected by an ancient curse, so I wouldn't want to disturb it too much.

File: lib/random_word.rb (hunk 2 above)
Comment (toddthomas) [inline lib/random_word.rb]: Also I forgot that `set_constraints` will default the values itself if passed no args, so I can delete more characters.

**Pull request merged**